### PR TITLE
Sap systems and Databases openapi

### PR DESCRIPTION
**Pull request by nelsonkopliku**
- [x] GET     /api/hosts
- [x] GET     /api/clusters
- [x] GET     /api/sap_systems
- [x] GET     /api/databases
- [ ] GET     /api/checks/catalog
- [ ] POST    /api/clusters/:cluster_id/checks
- [ ] POST    /api/clusters/:cluster_id/checks/request_execution
- [ ] POST    /api/runner/callback

For extra information see here 
https://github.com/trento-project/web/pull/512
https://github.com/trento-project/web/pull/522

### CODE REVIEW

File: lib/trento_web/openapi/schema/database.ex [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+  defmodule DatabaseInstance do
+    @moduledoc false
+
+    OpenApiSpex.schema(%{
+      title: "DatabaseInstance",
+      description: "A discovered HANA Database Instance on the target infrastructure",
+      type: :object,
+      properties: %{
+        sap_system_id: %Schema{type: :integer, description: "SAP System ID", format: :uuid},
```
Comment (arbulu89) [inline lib/trento_web/openapi/schema/database.ex:17]: Is this `type: :integer` correct? Shouldn't be `:string` as at the end it is a uuid

File: lib/trento_web/openapi/schema/database.ex [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        sid: %Schema{type: :string, description: "SID"},
+        tenant: %Schema{type: :string, description: "Tenant"},
+        instance_number: %Schema{type: :string, description: "Instance Number"},
+        instance_hostname: %Schema{type: :string, description: "Instance Hostname"},
+        features: %Schema{type: :string, description: "Instance Features"},
+        http_port: %Schema{type: :integer, description: "Instance HTTP Port"},
+        https_port: %Schema{type: :integer, description: "Instance HTTPS Port"},
+        start_priority: %Schema{type: :string, description: "Instance Start Priority"},
```
Comment (arbulu89) [inline lib/trento_web/openapi/schema/database.ex:25]: The instance priority is a number with decimals

File: lib/trento_web/openapi/schema/database.ex [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+    OpenApiSpex.schema(%{
+      title: "DatabaseInstance",
+      description: "A discovered HANA Database Instance on the target infrastructure",
+      type: :object,
+      properties: %{
+        sap_system_id: %Schema{type: :integer, description: "SAP System ID", format: :uuid},
+        sid: %Schema{type: :string, description: "SID"},
+        tenant: %Schema{type: :string, description: "Tenant"},
+        instance_number: %Schema{type: :string, description: "Instance Number"},
```
Comment (arbulu89) [inline lib/trento_web/openapi/schema/database.ex:20]: The instance number is a number usually, but as they add the initial `0` most of the times, I guess having as string is still valid

File: lib/trento_web/openapi/schema/database.ex [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+      }
+    })
+  end
+
+  defmodule DatabasesCollection do
+    @moduledoc false
+
+    OpenApiSpex.schema(%{
+      title: "DatabasesCollection",
+      description: "A list of the discovered SAP Systems",
```
Comment (arbulu89) [inline lib/trento_web/openapi/schema/database.ex:75]: `A list of the discovered HANA Databases` maybe better

File: lib/trento_web/openapi/schema/sap_system.ex [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+  defmodule ApplicationInstance do
+    @moduledoc false
+
+    OpenApiSpex.schema(%{
+      title: "ApplicationInstance",
+      description: "A discovered Application Instance on the target infrastructure",
+      type: :object,
+      properties: %{
+        sap_system_id: %Schema{type: :integer, description: "SAP System ID", format: :uuid},
```
Comment (arbulu89) [inline lib/trento_web/openapi/schema/sap_system.ex:17]: Type -> `:string`

File: lib/trento_web/openapi/schema/sap_system.ex [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+        sid: %Schema{type: :string, description: "SID"},
+        instance_number: %Schema{type: :string, description: "Instance Number"},
+        instance_hostname: %Schema{type: :string, description: "Instance Hostname"},
+        features: %Schema{type: :string, description: "Instance Features"},
+        http_port: %Schema{type: :integer, description: "Instance HTTP Port"},
+        https_port: %Schema{type: :integer, description: "Instance HTTPS Port"},
+        start_priority: %Schema{type: :string, description: "Instance Start Priority"},
```
Comment (arbulu89) [inline lib/trento_web/openapi/schema/sap_system.ex:24]: Number with decimals

File: lib/trento_web/openapi/schema/sap_system.ex [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+  defmodule SAPSystemItem do
+    @moduledoc false
+
+    OpenApiSpex.schema(%{
+      title: "SAPSystem",
+      description: "A discovered SAP System on the target infrastructure",
+      type: :object,
+      properties: %{
+        id: %Schema{type: :integer, description: "SAP System ID", format: :uuid},
```
Comment (arbulu89) [inline lib/trento_web/openapi/schema/sap_system.ex:43]: Type -> string

**Review by arbulu89 — CHANGES_REQUESTED** [review]
Hey @nelsonkopliku ,
Nice job. I have just found some missmatches in the types, some descriptions and tiny details.

File: lib/trento_web/openapi/schema/database.ex (hunk 2 above)
Comment (nelsonkopliku) [reply lib/trento_web/openapi/schema/database.ex:25]: Interesting, we're defining it as a string in commands, events and readmodels.

I'll keep it like this and create an item in the backlog if that's ok.

File: lib/trento_web/openapi/schema/database.ex (hunk 3 above)
Comment (nelsonkopliku) [reply lib/trento_web/openapi/schema/database.ex:20]: Yeah, true.

File: lib/trento_web/openapi/schema/database.ex (hunk 1 above)
Comment (nelsonkopliku) [reply lib/trento_web/openapi/schema/database.ex]: changed

File: lib/trento_web/openapi/schema/database.ex (hunk 2 above)
Comment (arbulu89) [reply lib/trento_web/openapi/schema/database.ex:25]: Totally fine! We don't do anything more than displaying it, so string is even good for us

**Review by arbulu89 — APPROVED** [review]
Go!

**Comment by nelsonkopliku** [comment]
> Go!

Elixir! :joy:

**Pull request merged**